### PR TITLE
Adding test cases and fetching information from DB for user profile

### PR DESCRIPTION
**Pull request by imran1289-ah**
This PR is related to #71  and user story #52 
- Fetch the information from the db and display in the front-end
- Write 1 test case in the front end to check if page is rendered correctly
- Write 1 test case in backend to test search functionality
- *I had to change some of the attributes in the model because i was having trouble with displaying them when they were arrays. I spent lots of time and couldn't find a solution and since we don't have much time left, I decided to change them as string and display them.

### CODE REVIEW

**Review by Kunal22shah — APPROVED** [review]
Everything looks good to me, with the limited time we have for this sprint we can try to debug the arrays issue in the user model in the next sprint!

File: src/client/src/components/PublicUserProfile.jsx [hunk 1]
```diff
@@ -11,18 +11,21 @@ const PublicUserProfile = () => {
...
-        const response = await axios.get(
-          `http://localhost:9000/users/profile/${profileId}`
-        );
-        setPublicUser(response.data);
-      } catch (error) {
-        console.error(error);
-      }
-    };
     fetchProfile();
   }, [locationURL]);
```
Comment (AnnaHsu1) [inline src/client/src/components/PublicUserProfile.jsx:30]: So the component re-renders when the URL changes?

**Review by AnnaHsu1 — COMMENTED** [review]
Would doing a for loop to display the array work? Each loop would create a new div containing the element. Otherwise, looks good!

**Review by AnnaHsu1 — APPROVED** [review]

File: src/client/src/components/PublicUserProfile.jsx (hunk 1 above)
Comment (imran1289-ah) [reply src/client/src/components/PublicUserProfile.jsx]: yes ? are you suggesting that it should depend on something else?

**Comment by imran1289-ah** [comment]
Fixed issue of displaying some of the attributes as arrays

File: src/client/src/components/PublicUserProfile.jsx (hunk 1 above)
Comment (AnnaHsu1) [reply src/client/src/components/PublicUserProfile.jsx]: No this makes sense, just wanted to confirm!

**Pull request merged**